bsb_nest/devices/sinusoidal_poisson_generator.py

```diff
@@ -0,0 +1,55 @@
+import nest
+from bsb import config
+from neo import SpikeTrain
+
+from ..device import NestDevice
+
+
+@config.node
+class SinusoidalPoissonGenerator(
+    NestDevice, classmap_entry="sinusoidal_poisson_generator"
+):
+    rate = config.attr(type=float, required=True)
+    """Rate of the poisson generator"""
+    amplitude = config.attr(type=float, required=True)
+    """Amplitude of the sinusoidal signal"""
+    frequency = config.attr(type=float, required=True)
+    """Frequency of the sinusoidal signal"""
+    phase = config.attr(type=float, required=False, default=0.0)
+    """Phase of the sinusoidal signal"""
+    start = config.attr(type=float, required=False, default=0.0)
+    """Activation time in ms"""
+    stop = config.attr(type=float, required=False, default=None)
+    """Deactivation time in ms. 
+        If not specified, generator will last until the end of the simulation."""
+
+    def implement(self, adapter, simulation, simdata):
+        nodes = self.get_target_nodes(adapter, simulation, simdata)
+        params = {
+            "rate": self.rate,
+            "start": self.start,
+            "amplitude": self.amplitude,
+            "frequency": self.frequency,
+            "phase": self.phase,
+        }
+        if self.stop is not None and self.stop > self.start:
```

Comment:
Throw an error if `self.stop <= self.start`, to help out people who made a typo ;) You could also deal with it in the type handler of the attribute to raise an early `ConfigurationError` rather than catching it here.